openapi.yaml

```diff
@@ -3576,11 +3598,15 @@ components:
       description: Goal template creation request.
       example:
         name: New template name
+        repositories: null
       properties:
         name:
           $ref: '#/components/schemas/GoalTemplateName'
+        repositories:
+          $ref: '#/components/schemas/GoalRepositories'
       required:
       - name
+      - repositories
```

Comment:
Not required.

Reply:
What is the semantic of not including `repositories` in the `update_goal_template` operation ?
The API should set the field to `null` ? Or the API should implement a partial update logic and not change the field ?
Do you have other examples of partial updates in the API ?

Reply:
I guess on creation it can be optional and be interpreted as `null` instead

Reply:
The request doesn't contain the fields that shouldn't be updated. This is how it's working for other endpoints and models. Also, we have a PUT that overwrites the whole goal, so we don't support partial updates. We don't support partial updates anywhere in the current endpoints (well, except maybe `PUT /account/{id}/features`, but it's arguable), and it's a good thing.